sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/ReactorNettyHttpResponseBase.java

```diff
@@ -31,8 +42,212 @@ public final String getHeaderValue(String name) {
+                        // caching is available.
+                        if (nettyHeaders.contains((String) key)) {
+                            return String.join(",", nettyHeaders.getAll((String) key));
+                        }
+                        return null;
+                    }
+
+                    @Override
+                    public String put(String key, String value) {
+                        throw new UnsupportedOperationException("HttpHeaders are unmodifiable");
+                    }
+
+                    @Override
+                    public String remove(Object key) {
+                        throw new UnsupportedOperationException("HttpHeaders are unmodifiable");
+                    }
+
+                    @Override
+                    public void putAll(Map<? extends String, ? extends String> m) {
+                        throw new UnsupportedOperationException("HttpHeaders are unmodifiable");
+                    }
+
+                    @Override
+                    public void clear() {
+                        throw new UnsupportedOperationException("HttpHeaders are unmodifiable");
+                    }
+
+                    @Override
+                    public Set<Entry<String, String>> entrySet() {
+                        return new AbstractSet<Entry<String, String>>() {
+                            @Override
+                            public Iterator<Entry<String, String>> iterator() {
+                                return nettyHeaders.iteratorAsString();
+                            }
+
+                            @Override
+                            public int size() {
+                                return nettyHeaders.size();
+                            }
+                        };
+                    }
+                };
+            }
+            return abstractMap;
+        }
+
+        @Override
+        public Iterator<HttpHeader> iterator() {
+            return stream().iterator();
+        }
+
+        @Override
+        public Stream<HttpHeader> stream() {
+            return StreamSupport.stream(nettyHeaders.spliterator(), false)
+                       .map(e -> new NettyHttpHeader(this, e));
+        }
+    }
+
+    static class NettyHttpHeader extends HttpHeader {
+        private final NettyHttpHeaders allHeaders;
+
+        NettyHttpHeader(NettyHttpHeaders allHeaders, String name, String value) {
+            super(name, value);
+            this.allHeaders = allHeaders;
+        }
+
+        NettyHttpHeader(NettyHttpHeaders allHeaders, Map.Entry<String, String> entry) {
+            this(allHeaders, entry.getKey(), entry.getValue());
+        }
+
+        NettyHttpHeader(NettyHttpHeaders allHeaders, String name, List<String> values) {
+            super(name, values);
+            this.allHeaders = allHeaders;
+        }
+
+        @Override
+        public void addValue(String value) {
```

Comment:
Don't think we need to support this update method as we are using this only for response headers.

Reply:
Will test by changing this to throw an exception and see what happens...

Reply:
It works, but as discussed offline, we will keep this functionality as-is, and allow mutability (even if it is unused).